app/src/main/java/cm/aptoide/pt/app/AppViewManager.java

```diff
@@ -346,12 +348,20 @@ public Completable downloadApp(DownloadModel.Action downloadAction, long appId,
             .flatMap(hasAds -> {
               if (hasAds) {
                 return moPubAdsManager.shouldShowAds()
-                    .doOnSuccess(showAds -> setupDownloadEvents(download, downloadAction, appId,
-                        trustedValue, editorsChoicePosition,
-                        showAds ? WalletAdsOfferManager.OfferResponseStatus.ADS_SHOW : ADS_HIDE));
+                    .doOnSuccess(showAds -> {
+                      setupDownloadEvents(download, downloadAction, appId, trustedValue,
+                          editorsChoicePosition,
+                          showAds ? WalletAdsOfferManager.OfferResponseStatus.ADS_SHOW : ADS_HIDE);
+                      if (downloadAction.equals(DownloadModel.Action.MIGRATE)) {
+                        setupMigratorUninstallEvent(download.getPackageName());
+                      }
+                    });
               } else {
                 setupDownloadEvents(download, downloadAction, appId, trustedValue,
                     editorsChoicePosition, NO_ADS);
+                if (downloadAction.equals(DownloadModel.Action.MIGRATE)) {
+                  setupMigratorUninstallEvent(download.getPackageName());
```

Comment:
Since this is not an uninstall event, I would change the name to something like setupAppcMigrationEvents(...)